### ISSUE TITLE
Add form validation to prevent links to preview.

### ISSUE THREAD

**Issue by swirtSJW** [department-of-veterans-affairs/va.gov-cms#10356]
## Description
TL;DR: A link to the preview server should be prevented from being created so that editors do not link to something that veterans can not get to.

Editors create content in Drupal, and this content is used to build the frontend. Unfortunately, editors can input links to local files on their machines, resources located inside our private networks, etc. The specific issue here is that editors can create links to the content-preview servers, which they use to preview their work as they edit it, and there is no validation in place at the moment to prevent it.

We have two independent content preview servers for staging and prod. Their hostnames are `preview-staging.vfs.va.gov` and `preview-prod.vfs.va.gov`. Tugboat and local environments use a different mechanism and aren't relevant here.

For prior art, see:
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteUrlsAsPathsInLinks.php`
- `docroot/modules/custom/va_gov_backend/src/Plugin/Validation/Constraint/PreventAbsoluteUrlsAsPathsInLinksValidator.php`
- (any other validator in the directory above)
- `docroot/modules/custom/va_gov_backend/va_gov_backend.module` (lines 1323, 1332)
- `tests/phpunit/Content/PreventAbsoluteUrlsAsPathsInLinksValidatorTest.php`

## Acceptance Criteria
- [ ] A link created to the preview server url will fail form validation and prevent save.
- [ ] Validator is covered by unit tests.
